lib/mirage/mirage.ml

```diff
@@ -485,6 +485,12 @@ let ( ++ ) acc x =
 
 let register ?(argv = default_argv) ?tracing ?(reporter = default_reporter ())
     ?keys:extra_keys ?packages ?src name jobs =
+  if List.exists Functoria.Impl.app_has_no_arguments jobs then
+    invalid_arg "Your configuration includes a job without arguments. \
+                 Please add a dependency in your config.ml: use \
+                 `let main = Mirage.main \"Unikernel.hello\" (noop @-> job) \
```

Comment:
Regarding the example later, it should be `let main = Mirage.main \"Unikernel.hello\" (job @-> job) \`, no?

Reply:
indeed, thanks for spotting this, fixed in 28e80385